server/engine/src/main/java/org/eclipse/lsp/cobol/dialects/ibm/ParserStage.java

```diff
@@ -89,4 +99,117 @@ private List<SyntaxError> getParsingError(AnalysisContext context, AstBuilder pa
   public String getName() {
     return "Parsing stage";
   }
+
+  private void processCobolJavaInteroperabilityDirectives(
+      AnalysisContext context, CommonTokenStream tokenStream) {
+    List<Token> compilerLineTokens =
+        tokenStream.getTokens().stream()
+            .filter(token -> token.getChannel() == Lexer.HIDDEN)
+            .filter(token -> token.getType() == CobolLexer.COMPILERLINE)
+            .collect(Collectors.toList());
```

Comment:
```suggestion
    List<Token> compilerLineTokens =
        tokenStream.getTokens().stream()
            .filter(token -> token.getType() == CobolLexer.COMPILERLINE || == IDENTIFICATION  ||  ==DATA || PROCEDURE)
            .collect(Collectors.toList());
```

and then it should be sufficient to check that each JAVA-CALLABLE is sandwiched between IDENTIFICATION  and PROCEDURE and each JAVA-SHAREABLE is sandwiched DATA and PROCEDURE.